File: abc144/abc144_c.cs
// https://atcoder.jp/contests/abc144/submissions/8160350

using System;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var N = long.Parse(Console.ReadLine());

            var ub = (int)Math.Floor(Math.Sqrt(N));
            for (int i = ub; i > 0; i--)
            {
                if (N % i == 0)
                {
                    var l = i;
                    var r = N / i;
                    Console.WriteLine(l + r - 2);
                    break;
                }
            }
        }
    }
}

File: abc/abc144/abc144_c.fs
// https://atcoder.jp/contests/abc144/submissions/12055089

let N = stdin.ReadLine() |> int64

let rec solver i =
    if N % i <> 0L then solver (i - 1L)
    else i + N / i - 2L

solver
    (N
     |> float
     |> sqrt
     |> floor
     |> int64)
|> stdout.WriteLine
